### ISSUE TITLE
[MEI_encoding] Syllable and neume contain components on either end of the staff--this is strange

### ISSUE THREAD

**Issue by JoyfulGen** [DDMAL/Rodan#784]
First, I think this is an MEI_encoding issue because this only started happening when I started processing folios with the newest version of that job. @sabrina0822, for record everything else works super smoothly so far! 

So here's what's happening. In the picture below, the green boys at the beginning of the central staff and the solitary green boy at the end belong to the same syllable (I'm highlighting by syllable in that image). Also, the symbols in pink boxes all belong to one neume and the symbols in the green box belong to another. The snippet of the MEI file is below, with matching boxes:
<img width="947" alt="240r mysterious syl folio" src="https://user-images.githubusercontent.com/83373378/193844873-9fbf30b6-2b8c-4149-a76f-98cf347d3334.png">
<img width="1115" alt="240r mysterious syl MEI" src="https://user-images.githubusercontent.com/83373378/193845029-3cfff847-80fd-4d54-9bcb-79c4e9038df4.png">

In the first image, you can also see that the staff below has an empty spot at the beginning where those neumes are clearly supposed to go. This looks a lot like this Neon issue https://github.com/DDMAL/Neon/issues/762, which had to do with toggled syllables. Could they be part of the problem? However, in this situation there are also two halves of a **neume** on either end of a staff, which has nothing to do with toggled syllables.

I don't have any more data for now, but I'll add to this issue as soon as I do.

This is the complete MEI file:
[240rneume_either_end_staff.mei.zip](https://github.com/DDMAL/Rodan/files/9707782/240rneume_either_end_staff.mei.zip)

**Comment by yinanazhou** [comment]
Hi, I found that somehow the third green boy is recognized as a separated syllable, and it is also in green by accident. So far I'm not sure this is related to toggle-linked syllables.

**Comment by JoyfulGen** [comment]
I have investigated this issue further and have an update:

The problem is NOT, as I first thought, that a neume has components on either ends of the staff. The problem IS that a neume has components on two different staves and Neon doesn't know how to deal with that. Here's an illustration, from folio 232v:
<img width="911" alt="Jumpy neume illustration 232v" src="https://user-images.githubusercontent.com/83373378/207639377-5280ee7e-bc5d-4f2b-a7c2-08352ed82298.png">
In the above screenshot, I'm highlighting by neume. Both blue friends with squares around them belong to the same neume. Blue friend and green friend belong to the same syllable. You can see that part A of the blue friend and the green friend both clearly come from the staff below; they're perfectly aligned vertically, they're just on the wrong staff. 

This is the straight-outta-OMR MEI file, _before_ uploading it to Neon (colours match the above screenshot):

<img width="1223" alt="Jumpy neume MEI 232v" src="https://user-images.githubusercontent.com/83373378/207639594-45ff90ff-95f7-4be1-85d8-7eff1966b11e.png">

You can see that part B of the blue friend appears _first_, because part A should be at the beginning of the staff below. However, since part A is part of the same neume as part B, it has to appear under the same syllable, so it's assigned to the staff above where it should be. The same goes for the green friend: it's part of the same syllable as part A, so it's assigned to the staff above also. This is what creates the situation where a neume appears to be split on either end of the staff: it's actually split between two staves, it's just that the MEI doesn't allow for that, so it compensates as it can.

Notice also the `precedes` and `follows` tags: blue friend precedes yellow friend, yellow friend follows blue friend _and_ precedes indigo friend (even though they're on the same staff) and indigo friend also follows blue friend. When the file is uploaded to Neon, all those tags disappear.

I can't explain why this is happening, but there's a definite pattern when it happens. The generic MEI always looks like this (I hope this is clear, I did my best):
```
<Syllable 1> - `precedes` Syllable 2
    <Neume>
        First half of the neume (end of staff 1)
        Second half of the neume (should be second syllable of staff 2.)
    <Neume>
        Other neumes that should be on staff 2, but are appearing directly above on staff 1
<Syllable>
sb
<Syllable 2> First syllable of staff 2 - `follows` syllable 1, `precedes` syllable 3 (even though they’re on the same staff)
<Syllable 3> Last syllable of staff 2 (if it exists) - `follows` syllable 1
sb
```

This is the straight-outta-OMR file for 232v: 
[232v.mei.zip](https://github.com/DDMAL/Rodan/files/10229396/232v.mei.zip)
And this is the file after uploading it to Neon:
[232v after Neon.mei.zip](https://github.com/DDMAL/Rodan/files/10229410/232v.after.Neon.mei.zip)

Please let me know if you need anything else, of if I can explain something better!

**Comment by sabrina0822** [comment]
Quick question for @yinanazhou since I haven't worked with neon much before: even though the precedes and follows tags aren't present in xml files for Neon are the syllables still linked?

**Comment by yinanazhou** [comment]
Hi, I'm not sure I understand your question, but the `@precedes` and `@follows` tags should point to the `@xml:id` of each other, which indicates the two syllables are linked. This only (or usually?) happens when the syllable is separated by a staff break and use the tags to link them back. Please let me know if you have any questions.

**Comment by sabrina0822** [comment]
In the new build_MEI file I used the `precedes` and `follows` tags when a syllable was split up by a line break, clef, accidental, etc. Like this: 
```
<syllable xml-id:12345>
     <syl>
     <neume>
         <nc>
      <\neume>
     <accid>
     <neume>
         <nc>
         <nc>
      <\neume>
<\syllable>
```
becomes
```
<syllable xml-id:12345 precedes:67890>
     <syl>
     <neume>
         <nc>
      <\neume>
<\syllable>
<accid>
<syllable xml-id:67890 follows:12345>
     <neume>
         <nc>
         <nc>
      <\neume>
<\syllable>
```
@JoyfulGen this is why they are split weirdly in the xml file. 

These tags aren't seen in the xml file when uploaded to Neon and I was wondering if they were just handled differently @yinanazhou (this the file gen screenshotted in the first post after uploaded to neon).

<img width="893" alt="image" src="https://user-images.githubusercontent.com/90227881/208798647-ef817f41-9d86-4785-9574-899c0d9aa61a.png">

**Comment by sabrina0822** [comment]
I'm kinda stumped by this to be completely honest. 

What I think is happening is that an `<sb>` (indicates a new staff line is beginning) is happening in the middle of the first neume of syllable 1 (between B and A) and the code that I wrote doesn't have a way to handle it.

I'm guessing that as a result the neumes continue as usual and are added to syllable 1 and the `<sb>` happens after. To be honest this was an edge case I hadn't thought of because I didn't know it happened. I'm not 100% sure that this is what is happening but it would make sense given the way the mei file is split up. I'll be able to tell more clearly once I get the inputs to the mei_encoding job. 

In this case would it be best to split up the neumes into neume B and neume A? Not sure if you're allowed to do that but I don't how to handle it otherwise. In mei_v5 `<sb>`'s aren't allowed to occur in a `<neume>` so it would have to be in the `<syllable>` element. 

I tried my best to be clear but I'm not sure if this makes sense so let me know if you guys need any clarification :)

**Comment by JoyfulGen** [comment]
> Are the two blue friends supposed to be in the same neume or a different one?

No they are not! There are two things that are wrong in this situation, actually:
1. Syllables can cross a staff break, but neumes don't. So there shouldn't be any situation where neume components on two different staves are part of the same neume.
2. Even if it were possible for neumes to cross staves, this would still be wrong, because (in the screenshot example above) the yellow friend is between the two halves of the blue friend and non-consecutive neume components can't be part of the same neume.

So I guess something somewhere is allowing neume components on different staves to be part of the same neume, but that should never happen in the first place. 

A small additional detail: in the situations where this happens, it's always the **last** neume components of a staff that are grouped with the **second** neume component(s) of the staff following (like in the example above, where you have the yellow friend before the second half of the blue friend). I hope that's useful!

**Comment by yinanazhou** [comment]
I just checked the code, Neon does add @precedes and @follows tags to the MEI file if there are neumes before and after a system break [here](https://github.com/DDMAL/Neon/blob/b1d9c05230807bfa62aef8e3e395f07882554ca2/src/utils/ConvertMei.ts#L165). I will double check if Neon ever removes the tags automatically. I will need to play with the MEI files a bit to learn what is happening here. @JoyfulGen, could you please upload the corresponding image file for this folio?

**Comment by yinanazhou** [comment]
> In the new build_MEI file I used the `precedes` and `follows` tags when a syllable was split up by a line break, clef, accidental, etc. 

One thing I noticed here is that clef and accidental can be inside a syllable, and these elements do not necessarily separate a syllable. Linked syllables only link syllables separated by a staff break.

**Comment by yinanazhou** [comment]
Another thing I noticed is that in the straight-outta-omr MEI file, it should be `@precedes` instead of `@xml:precedes`. Reference here: https://music-encoding.org/guidelines/dev/elements/syllable.html#:~:text=separated%20by%20spaces.-,precedes,-(optional)Points

**Comment by fujinaga** [comment]
Have @timothydereuse fix that, if you cannot.

**Comment by sabrina0822** [comment]
@yinanazhou oh you're right I'll fix those two things

**Comment by fujinaga** [comment]
Thank you, @sabrina0822 !

**Comment by JoyfulGen** [comment]
> I just checked the code, Neon does add @precedes and @follows tags to the MEI file if there are neumes before and after a staff break [here](https://github.com/DDMAL/Neon/blob/b1d9c05230807bfa62aef8e3e395f07882554ca2/src/utils/ConvertMei.ts#L165). I will double check if Neon ever removes the tags automatically. I will need to play with the MEI files a bit to learn what is happening here. @JoyfulGen, could you please upload the corresponding image file for this folio?

![Einsiedeln__Stiftsbibliothek__Codex_611_232v](https://user-images.githubusercontent.com/83373378/209344702-7d30bc7c-0811-4669-8636-ba4e3a7e2295.jpg)

**Comment by sabrina0822** [comment]
> > Are the two blue friends supposed to be in the same neume or a different one?
> 
> No they are not! There are two things that are wrong in this situation, actually:
> 
> 1. Syllables can cross a staff break, but neumes don't. So there shouldn't be any situation where neume components on two different staves are part of the same neume.
> 2. Even if it were possible for neumes to cross staves, this would still be wrong, because (in the screenshot example above) the yellow friend is between the two halves of the blue friend and non-consecutive neume components can't be part of the same neume.
> 

Just wondering, in the instances where syllables cross a staff break are the two halves of the syllable seperated by a precedes and follows?

**Comment by JoyfulGen** [comment]
I'm quite sure that yes, they are, but @yinanazhou can confirm.

**Comment by yinanazhou** [comment]
Hi @sabrina0822, I've attached an example of separated syllables:
![image](https://user-images.githubusercontent.com/74484576/213944288-2ac1f9f9-dcf8-4d95-bca6-87ac7516990b.png)

**Comment by yinanazhou** [comment]
I've checked with the 232v file provided by Gen. In this case, Neon didn't change the file at all since it only recognises `precedes` or `follows`, not `xml: precedes` or `xml: follows`. 

@sabrina0822, another thing I noticed that should never happen is that the second syllable has `xml: precedes` and `xml: follows` at the same time. Neon would freak out if this happens. (I will add a check for this to Neon)
![image](https://user-images.githubusercontent.com/74484576/213945499-480ca42e-ae5a-4f33-a3fe-83682eb9828f.png)
